### PR TITLE
Add sorting

### PR DESCRIPTION
**Pull request by mimo00**
This change is [Reviewable](https://reviewable.io/reviews/socialwifi/flask-jsonapi/33)

### CODE REVIEW

File: flask_jsonapi/resource_repositories/sqlalchemy_repositories.py [hunk 1]
```diff
@@ -91,6 +92,14 @@ def apply_pagination(self, query, pagination):
         else:
             return query
 
+    def apply_sorts(self, query, sorting):
+        if sorting:
+            try:
+                query = query.order_by(*sorting)
+            except exc.InvalidRequestError:
+                raise exceptions.InvalidSort("{} has invalid sorts {}".format(self.model.__name__, sorting))
```
Comment (maruqu) [inline flask_jsonapi/resource_repositories/sqlalchemy_repositories.py]: Invalid sort fields for self.instance_name

File: flask_jsonapi/resource_repositories/sqlalchemy_repositories.py [hunk 2]
```diff
@@ -91,6 +92,14 @@ def apply_pagination(self, query, pagination):
         else:
             return query
 
+    def apply_sorts(self, query, sorting):
+        if sorting:
```
Comment (maruqu) [inline flask_jsonapi/resource_repositories/sqlalchemy_repositories.py]: you can use similar pattern as above:

         sorting = sorting or []

File: flask_jsonapi/filters_schema.py [hunk 3]
```diff
@@ -152,6 +152,14 @@ def get_fields(cls):
             raise ValueError('`fields` option must be a list or tuple.')
         return cls._meta.fields
 
+    @staticmethod
+    def get_attribute(schema, field):
```
Comment (maruqu) [inline flask_jsonapi/filters_schema.py]: get_model_attribute will be more informative

File: flask_jsonapi/query_string.py [hunk 4]
```diff
@@ -114,3 +116,29 @@ def extract_fields(self, value):
...
+            sort_fields = flask.request.args.get('sort').split(',')
+            sorting_results = []
+            for sort_field in sort_fields:
+                result = self.get_sorting(sort_field)
+                sorting_results.append(result)
+            return sorting_results
+        except AttributeError:
+            return []
+
+    def get_sorting(self, sort_field) -> str:
```
Comment (maruqu) [inline flask_jsonapi/query_string.py]: maybe get_model_sort_field ?

File: flask_jsonapi/query_string.py [hunk 5]
```diff
@@ -114,3 +116,29 @@ def extract_fields(self, value):
...
+
+    def parse(self) -> typing.List[str]:
+        try:
+            sort_fields = flask.request.args.get('sort').split(',')
+            sorting_results = []
+            for sort_field in sort_fields:
+                result = self.get_sorting(sort_field)
+                sorting_results.append(result)
+            return sorting_results
+        except AttributeError:
```
Comment (maruqu) [inline flask_jsonapi/query_string.py]: what this except is for?  better not to fail silently

File: flask_jsonapi/resource_repositories/sqlalchemy_repositories.py (hunk 2 above)
Comment (mimo00) [reply flask_jsonapi/resource_repositories/sqlalchemy_repositories.py]: Done

File: flask_jsonapi/resource_repositories/sqlalchemy_repositories.py (hunk 1 above)
Comment (mimo00) [reply flask_jsonapi/resource_repositories/sqlalchemy_repositories.py]: Done

File: flask_jsonapi/filters_schema.py (hunk 3 above)
Comment (mimo00) [reply flask_jsonapi/filters_schema.py]: Done

File: flask_jsonapi/query_string.py (hunk 4 above)
Comment (mimo00) [reply flask_jsonapi/query_string.py]: Done

File: flask_jsonapi/query_string.py (hunk 5 above)
Comment (mimo00) [reply flask_jsonapi/query_string.py]: Done

**Review by maruqu — COMMENTED** [review]
> Reviewed 6 of 7 files at r2, 1 of 6 files at r5, 3 of 3 files at r6, 2 of 2 files at r7, 4 of 4 files at r8.
> *[Reviewable](https://reviewable.io/reviews/socialwifi/flask-jsonapi/33)* status: all files reviewed, 5 unresolved discussions (waiting on @maruqu)

**Review by maruqu — COMMENTED** [review]
:lgtm_strong: 

> *[Reviewable](https://reviewable.io/reviews/socialwifi/flask-jsonapi/33#-:-LKaW1R31zYQAezrP4zc:b-uwkowd)* status: all files reviewed, 5 unresolved discussions (waiting on @maruqu)

**Review by maruqu — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/socialwifi/flask-jsonapi/33)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**